### DISCUSSION TITLE
Query automation on same schema, different table names

### DISCUSSION THREAD

**Discussion by TayPark** [prisma/prisma#8745] [Q&A] [answered]
There're 51 tables in my database, 3 schemas, 17 tables with different name use one schema.(3 * 17 = 51). Also there're table prefix and postfix, same prefix shares schema, tablename differ with postfix.

I'm gonna execute upsert query like `prismaClient[tableName].upsert()`. But prisma can upsert when tablename and upsert data schema are same. How can I automate it?

Here's code.

```typescript
// types.ts

type ValueOf<T> = T[keyof T]

export type TSido =
  | '서울특별시'
  | '인천광역시'
  | '대전광역시'
  | '대구광역시'
  | '광주광역시'
  | '울산광역시'
  | '부산광역시'
  | '세종특별자치시'
  | '제주특별자치도'
  | '경기도'
  | '강원도'
  | '충청남도'
  | '충청북도'
  | '전라남도'
  | '전라북도'
  | '경상남도'
  | '경상북도'

export type TSidoEng =
  | 'seoul'
  | 'incheon'
  | 'daejeon'
  | 'daegu'
  | 'ulsan'
  | 'busan'
  | 'gwangju'
  | 'sejong'
  | 'jeju'
  | 'gyeonggi'
  | 'gangwon'
  | 'chungnam'
  | 'chungbuk'
  | 'jeonnam'
  | 'jeonbuk'
  | 'gyeongnam'
  | 'gyeongbuk'

export type TSidoObject = {
  [key in TSido]: TSidoEng
}

export const SidoObject: TSidoObject = {
  서울특별시: 'seoul',
  인천광역시: 'incheon',
  대전광역시: 'daejeon',
  대구광역시: 'daegu',
  광주광역시: 'gwangju',
  울산광역시: 'ulsan',
  부산광역시: 'busan',
  세종특별자치시: 'sejong',
  제주특별자치도: 'jeju',
  경기도: 'gyeonggi',
  강원도: 'gangwon',
  충청남도: 'chungnam',
  충청북도: 'chungbuk',
  전라남도: 'jeonnam',
  전라북도: 'jeonbuk',
  경상남도: 'gyeongnam',
  경상북도: 'gyeongbuk',
}

export type TTablePrefix = 'additional_info' | 'roadname_address' | 'jibun_address'


// tablename is combine with TablePrefix and ValueOf TSidoObject. 
export type TTableName = `${TTablePrefix}_${ValueOf<TSidoObject>}`

```

```typescript
// updateQuery.ts

const updateData = '
let tablePrefix = ''
let tablePostfix = ''

// get table prefix and postfix by file contents. In this case, use 'additional_info' and 'seoul'.
talblePostfix = 'additional_info'
tablePrefix = 'seoul'

// combine it
const tableName = tablePrefix + '_' + tablePostfix as TTableName

// query exec
await prismaClient[tableName].upsert({
                                                   // ^^^^^
  where: { manage_number: addInfo.manage_number },
  update: addInfo,
  create: addInfo,
})
```

![image](https://user-images.githubusercontent.com/31007427/129437935-8f2ad36b-5e82-423d-bde3-de92aab079a2.png)

How do I fix it?

**Comment by ryands17** [comment, marked as answer]
@TayPark 👋 

Unfortunately this isn't easy to fix. We will be introducing a CRUD wrapper in the distant future and you can also check the comments here that specify a couple of approaches:
https://github.com/prisma/prisma/issues/5273

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
